src/com/jonfreer/wedding/application/services/GuestService.java

```diff
@@ -206,17 +207,23 @@ public int insertGuest(com.jonfreer.wedding.servicemodel.Guest guest) {
         }
     }
 
-    public ArrayList<com.jonfreer.wedding.servicemodel.Guest> getGuests() {
+    public ArrayList<com.jonfreer.wedding.servicemodel.Guest> getGuests(GuestSearchCriteria searchCriteria) {
 
         IDatabaseUnitOfWork unitOfWork =
-                this.databaseUnitOfWorkFactory.create();
+            this.databaseUnitOfWorkFactory.create();
         IGuestRepository guestRepository =
-                this.guestRepositoryFactory.create(unitOfWork);
+            this.guestRepositoryFactory.create(unitOfWork);
         IReservationRepository reservationRepository =
-                this.reservationRepositoryFactory.create(unitOfWork);
+            this.reservationRepositoryFactory.create(unitOfWork);
 
         try {
-            ArrayList<com.jonfreer.wedding.domain.Guest> guests = guestRepository.getGuests();
+            com.jonfreer.wedding.domain.GuestSearchCriteria searchCriteriaDomain = null;
+            if(searchCriteria != null){
```

Comment:
This check is necessary because `Dozer` throws an exception when the source object being mapped is `null`.